[Go] Implement delete user handler

katsurada1 opened the pull request:
## Issue Number
#675 

## Implementation Summary
Implement delete user handler.

## Scope of Impact
Affects OpenAPI definitions related to /api/users/{userID} endpoints.

## Particular points to check
Verify parameter handling `userID` in generated server interface and handlers.

## Test
What tests are included?

## Schedule
4/25

Amoonguses1 commented:
@katsurada1 
The transition from dockertest to fake requires all handlers to be fake. Therefore, feature/#661_refactor_backend_tests is used as a dev branch. Please switch the parent branch from main to the dev branch.

katsurada1 commented:
@Amoonguses1 
In oapi-codegen, if a path parameter defined in a root path (e.g., /api/users/{userID}) is not explicitly declared at the path level, the following error occurs:

```
error generating code: path '/api/users/{userID}' has 1 positional parameters, but spec has 0 declared
```

This is because oapi-codegen does not correctly interpret parameters defined at the operation level (within individual methods). According to the documentation and community reports, this is a known limitation.

Therefore, as a workaround in this case:
 - I moved the parameters declaration to the path level.
 - Accordingly, we removed the parameters section from find_user_by_id.yml.

This approach remains compliant with the OpenAPI specification while also addressing the limitations of oapi-codegen.

Please check my code.
